lib/web/apiserver.go

```diff
@@ -361,14 +361,38 @@ func (m *Handler) getAuthenticationSettings(w http.ResponseWriter, r *http.Reque
 	return pr, nil
 }
 
+type webConfig struct {
+	Auth interface{} `json:"auth"`
+	// ServerVersion is the version of Teleport that is running.
+	ServerVersion string `json:"serverVersion"`
+}
+
 // getConfigurationSettings returns configuration for the web application.
 func (m *Handler) getConfigurationSettings(w http.ResponseWriter, r *http.Request) (interface{}, error) {
-	pr, err := buildAuthenticationSettings(m.cfg.ProxyClient)
-	if err != nil {
-		return nil, trace.Wrap(err)
+	webCfg := webConfig{ServerVersion: teleport.Version}
+	as := client.AuthenticationSettings{}
+	cap, err := m.cfg.ProxyClient.GetClusterAuthPreference()
+
+	if err == nil {
```

Comment:
Yeah, it was a quick fix from my side to unblock myself. We should def. remove code duplication. 

I am leaving it up to you to address it upon your next changes if that's OK with you.

Please note that as far as config.js is concern, the serverVersion should not be part of the auth setttings. In case if invalid auth configuration, serverVersion wont be passed the client. That's why I am handling it  separately from auth params ->

webCfg := webConfig{ServerVersion: teleport.Version}